src/main/kotlin/lotto/domain/LottoMachine.kt

```diff
@@ -0,0 +1,57 @@
+package lotto.domain
+
+import lotto.data.Lotto
+import lotto.data.LottoNumber
+import lotto.data.LottoRanking
+import lotto.data.LottoRanking.FifthPlace
+import lotto.data.LottoRanking.FirstPlace
+import lotto.data.LottoRanking.FourthPlace
+import lotto.data.LottoRanking.None
+import lotto.data.LottoRanking.SecondPlace
+import lotto.data.LottoRanking.ThirdPlace
+import lotto.data.WinningLotto
+
+object LottoMachine {
+
+    fun createSelectLotto(lottoNumbers: Set<LottoNumber>): Lotto {
+        return Lotto(lottoNumbers)
+    }
+
+    fun createWinningLotto(winningNumbers: List<Int>, bonusNumber: Int): WinningLotto {
+        val winningLotto = LottoNumber.createLottoNumbers(winningNumbers)
+        val bonusLottoNumber = LottoNumber.from(bonusNumber)
+        return WinningLotto(Lotto(winningLotto), bonusLottoNumber)
+    }
+
+    fun checkLotto(purchaseLotto: Lotto, winningLotto: WinningLotto): LottoRanking {
+        val intersectNumber = winningLotto.lotto.selectNumbers.intersect(purchaseLotto.selectNumbers)
+
+        return when (intersectNumber.size) {
+            FirstPlace.matchingNumberCnt -> FirstPlace
+            SecondPlace.matchingNumberCnt, ThirdPlace.matchingNumberCnt -> {
+                checkSecondPlace(purchaseLotto, winningLotto.bonusNumber)
+            }
+            FourthPlace.matchingNumberCnt -> FourthPlace
+            FifthPlace.matchingNumberCnt -> FifthPlace
+            else -> None
+        }
```

Comment:
열거타입을 좀 더 활용해봐도 좋을 것 같아요. 
```suggestion
        return LottoRanking.findByMatchCnt(intersectNumber.size)
```

Reply:
WinningLotto에서 일치하는 번호 개수와 보너스 번호 일치 여부에 대한 값을 LottoRanking에 생성된 함수(findLottoRanking)에 넘겨주는 방식으로 수정하였습니다.
[아래 리뷰](https://github.com/next-step/kotlin-lotto/pull/959#discussion_r1398423745)의 보너스 번호에 따른 반환도 LottoRanking에서 처리하게 구성하였습니다.